src/main/java/core/basesyntax/stategy/handler/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.stategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer fruitQuantity = Storage.fruits.get(transaction.getFruitName());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity > 0 && purchaseQuantity <= fruitQuantity) {
```

Comment:
The condition should also include a check for 'fruitQuantity' being null, which signifies that the fruit is not present in the Storage. In such a case, a RuntimeException should be thrown as it is not possible to purchase a non-existing fruit.